lib/src/analysis_options.dart

```diff
@@ -29,63 +30,63 @@ class AnalysisOptions {
     try {
       final node = loadYamlNode(content ?? '');
 
-      return AnalysisOptions.fromYamlMap(node is YamlMap ? node : YamlMap());
+      return AnalysisOptions.fromMap(
+          node is YamlMap ? yamlMapToDartMap(node) : {});
     } on YamlException catch (e) {
       throw FormatException(e.message, e.span);
     }
   }
 
-  factory AnalysisOptions.fromYamlMap(YamlMap node) {
+  factory AnalysisOptions.fromMap(Map<String, Object> configMap) {
     Config metricsConfig;
-    var metricsExcludePatterns = <String>[];
+    Iterable<String> metricsExcludePatterns = <String>[];
     var rules = <String, Map<String, Object>>{};
 
-    if (node.nodes[_rootKey] is YamlMap) {
-      final metricsOptions = node.nodes[_rootKey] as YamlMap;
-
-      if (_isYamlMapOfStringsAndIntegers(metricsOptions.nodes[_metricsKey])) {
-        final configMap = metricsOptions[_metricsKey] as YamlMap;
+    final metricsOptions = configMap[_rootKey];
+    if (metricsOptions != null && metricsOptions is Map<String, Object>) {
```

Comment:
`null is T` returns false (which is kinda strange given that null is member of everytype), so there is no need for dedicated null-check.

Reply:
fixed